dali/ft/filecopy.cpp

```diff
@@ -1219,19 +1210,66 @@ void FileSprayer::calculateSprayPartition()
     if ((partitioners.ordinality() > 0) && !srcAttr->hasProp("ECL"))
     {
         // Store discovered CSV record structure into target logical file.
-        StringBuffer recStru;
-        partitioners.item(0).getRecordStructure(recStru);
-        if ((recStru.length() > 0) && strstr(recStru.str(),"END;"))
-        {
-            if (distributedTarget)
-                distributedTarget->setECL(recStru.str());
-        }
+        StoreCsvRecordStructure(partitioners.item(0));
     }
     if (compressedInput && compressOutput && streq(encryptKey.str(),decryptKey.str()))
         copyCompressed = true;
 
 }
 
+void FileSprayer::StoreCsvRecordStructure(IFormatPartitioner &partitioner)
+{
+    StringBuffer recStru;
+    partitioner.getRecordStructure(recStru);
+    if ((recStru.length() > 0) && strstr(recStru.str(),"END;"))
+    {
+        if (distributedTarget)
+            distributedTarget->setECL(recStru.str());
+    }
+}
+
+IFormatPartitioner * FileSprayer::createPartitioner(aindex_t index, bool calcOutput, unsigned numParts)
+{
+    StringBuffer remoteFilename;
+    FilePartInfo & cur = sources.item(index);
+    cur.filename.getRemotePath(remoteFilename.clear());
+    LOG(MCdebugInfoDetail, job, "Partition %d(%s)", index, remoteFilename.str());
+
+    srcFormat.quotedTerminator = options->getPropBool("@quotedTerminator", true);
+    const SocketEndpoint & ep = cur.filename.queryEndpoint();
+    IFormatPartitioner * partitioner = createFormatPartitioner(ep, srcFormat, tgtFormat, calcOutput, queryFixedSlave(), wuid);
+
+    // CSV record structure discovery of the first source
+    bool isRecordStructurePresent = options->getPropBool("@recordStructurePresent", false);
+    partitioner->setRecordStructurePresent(isRecordStructurePresent);
+
+    RemoteFilename name;
+    name.set(cur.filename);
+    setCanAccessDirectly(name);
+    partitioner->setPartitionRange(totalSize, cur.offset, cur.size, cur.headerSize, numParts);
+    partitioner->setSource(index, name, compressedInput, decryptKey);
+
+    return partitioner;
+}
+
+void FileSprayer::ExamineCsvStructure()
```

Comment:
See other comment, re. moving check for if ((partitioners.ordinality() > 0) && !srcAttr->hasProp("ECL")).
This method can add the partiioner (to standard 'partiioners' array) if necessary only (not call storeCsvRecordStructure)